### PR TITLE
[wasm] Wasm Websocket support

### PR DESCRIPTION
**Pull request by kjpou1**
Provides a `ClientWebSocket` implementation wrapping the browser `websocket` API.

- Connect to socket
- Send and Receive support for Text and Binary.  The browser WebSocket provides message-based API.
- Close socket.

issue: https://github.com/mono/mono/issues/9567

### CODE REVIEW

File: sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			string invalidChar = null;
+			int i = 0;
+			while (i < chars.Length) {
+				char ch = chars [i];
+				if (ch < 0x21 || ch > 0x7e) {
+					invalidChar = string.Format (CultureInfo.InvariantCulture, "[{0}]", (int)ch);
+					break;
+				}
+
+				if (!char.IsLetterOrDigit (ch) &&
```
Comment (kg) [inline sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs]: The IsLetterOrDigit check is redundant

File: sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		{
+			// We allow the subProtocol to be null. Validate if it is not null.
+			if (subProtocol != null) {
+				ValidateSubprotocol (subProtocol);
+			}
+
+			if (keepAliveInterval < Timeout.InfiniteTimeSpan) // -1
+			{
+				throw new ArgumentOutOfRangeException (nameof (keepAliveInterval), keepAliveInterval,
+				    $"Argument specified '{Timeout.InfiniteTimeSpan.ToString ()}' is too small.");
```
Comment (kg) [inline sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs]: Shouldn't the 'argument specified' bit be {keepAliveInterval} and not InfiniteTimeSpan?

File: sdks/wasm/sdk/WebAssembly.Net.WebSockets/WebAssembly.Net.WebSockets.csproj [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <Description>Mono.WebAssembly ClientWebSocket implementation. This package is not intended to be referenced directly.</Description>
+    <IncludeBuildOutput>false</IncludeBuildOutput>
+    <!-- get rid of warnings about assemblies not in lib -->
+    <NoPackageAnalysis>true</NoPackageAnalysis>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Include="..\..\..\wasm\framework\WebAssembly.Net.WebSockets.dll" PackagePath="lib\netstandard2.0\%(Filename)%(Extension)" Pack="True" />
```
Comment (kg) [inline sdks/wasm/sdk/WebAssembly.Net.WebSockets/WebAssembly.Net.WebSockets.csproj]: Maybe just use a wildcard here, WebAssembly.Net.WebSockets.*

File: sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs (hunk 1 above)
Comment (kjpou1) [reply sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs]: This is taken care of in recent commits

File: sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs (hunk 2 above)
Comment (kjpou1) [reply sdks/wasm/framework/src/WebAssembly.Net.WebSockets/WebSocketHelpers.cs]: good catch.  Recent commits now take this validation out so it should be resolved.

File: sdks/wasm/sdk/WebAssembly.Net.WebSockets/WebAssembly.Net.WebSockets.csproj (hunk 3 above)
Comment (kjpou1) [reply sdks/wasm/sdk/WebAssembly.Net.WebSockets/WebAssembly.Net.WebSockets.csproj]: I will be changing this soon.  Thanks for the heads up.

Comment (kjpou1) [reply sdks/wasm/sdk/WebAssembly.Net.WebSockets/WebAssembly.Net.WebSockets.csproj]: Should be modified now

**Pull request merged**